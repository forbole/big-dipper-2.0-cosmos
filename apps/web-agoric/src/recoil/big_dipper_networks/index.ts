export * from 'ui/recoil/big_dipper_networks';
