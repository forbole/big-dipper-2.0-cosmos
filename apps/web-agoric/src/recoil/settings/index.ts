export * from 'ui/recoil/settings';