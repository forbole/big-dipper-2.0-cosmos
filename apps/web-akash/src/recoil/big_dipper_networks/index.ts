export * from '@recoil/big_dipper_networks';
