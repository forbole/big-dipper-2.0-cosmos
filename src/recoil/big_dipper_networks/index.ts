import {
  atomState,
} from './atom';

import {
  useBigDipperNetworksRecoil,
} from './hooks';

import {
  writeNetworks,
  writeSelectedNetwork,
  readNetworks,
  readSelectedNetwork,
} from './selectors';

export {
  atomState,
  useBigDipperNetworksRecoil,
  writeNetworks,
  writeSelectedNetwork,
  readNetworks,
  readSelectedNetwork,
};
