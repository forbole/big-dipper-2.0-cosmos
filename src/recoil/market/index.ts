import {
  atomState,
} from './atom';

import {
  useMarketRecoil,
} from './hooks';

import {
  readMarket,
  writeMarket,
} from './selectors';

export {
  atomState,
  useMarketRecoil,
  readMarket,
  writeMarket,
};
