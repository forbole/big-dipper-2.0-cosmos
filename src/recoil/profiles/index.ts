import {
  atomFamilyState,
} from './atom';

import {
  useProfileRecoil,
  useProfilesRecoil,
} from './hooks';

import {
  readProfile,
  readProfiles,
  writeProfile,
  readDelegatorAddress,
  readDelegatorAddresses,
  readProfileExist,
  readProfilesExist,
  validatorToDelegatorAddress,
} from './selectors';

export {
  validatorToDelegatorAddress,
  atomFamilyState,
  useProfileRecoil,
  useProfilesRecoil,
  readProfile,
  readProfiles,
  writeProfile,
  readDelegatorAddress,
  readDelegatorAddresses,
  readProfileExist,
  readProfilesExist,
};
