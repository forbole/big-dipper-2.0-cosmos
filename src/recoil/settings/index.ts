import {
  atomState,
} from './atom';

import {
  writeTheme,
  readTheme,
  writeDate,
  readDate,
  writeTx,
  readTx,
} from './selectors';

import {
  THEME_LIST,
  THEME_DICTIONARY,
  DATE_LIST,
  TX_LIST,
  getThemeTemplate,
} from './utils';

import {
  useSettingsRecoil,
} from './hooks';

export {
  THEME_LIST,
  THEME_DICTIONARY,
  DATE_LIST,
  TX_LIST,
  useSettingsRecoil,
  atomState,
  writeTheme,
  getThemeTemplate,
  readTheme,
  writeDate,
  readDate,
  writeTx,
  readTx,
};
