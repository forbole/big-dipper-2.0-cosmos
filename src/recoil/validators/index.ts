import {
  atomFamilyState,
} from './atom';

import {
  useValidatorRecoil,
} from './hooks';

import {
  readValidator,
  writeValidator,
} from './selectors';

export {
  atomFamilyState,
  useValidatorRecoil,
  readValidator,
  writeValidator,
};
