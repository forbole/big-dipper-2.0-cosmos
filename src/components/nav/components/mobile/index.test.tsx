import React from 'react';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';
import {
  MockTheme, wait,
} from '@tests/utils';
import Mobile from '.';
// ==================================
// global setup
// ==================================
let component:renderer.ReactTestRenderer;

// ==================================
// global setup
// ==================================
jest.mock('./components', () => ({
  Menu: (props) => <div id="menu" {...props} />,
  // Network: (props) => <div id="network" {...props} />,
  Navbar: (props) => <div id="navbar" {...props} />,
  SearchBar: (props) => <div id="searchBar" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Nav/Mobile', () => {
  beforeEach(async () => {
    component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <Mobile title="hello world" />
        </MockTheme>
      </RecoilRoot>,
    );
    await wait();
  });

  it('it renders', async () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
