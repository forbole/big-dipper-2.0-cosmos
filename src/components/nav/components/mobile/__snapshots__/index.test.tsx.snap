// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/Mobile it renders 1`] = `
<div>
  <div
    className="makeStyles-root"
  >
    <div
      className="makeStyles-screens"
      id="menu"
      toggleNavMenus={[Function]}
    />
    <span
      className="makeStyles-screens"
    >
      <div
        className="makeStyles-networks"
      />
    </span>
    <div
      id="navbar"
      isOpen={false}
      openNetwork={[Function]}
      toggleNavMenus={[Function]}
    />
    <div
      className="makeStyles-searchBar"
      id="searchBar"
    />
  </div>
  <div
    style={
      Object {
        "height": 0,
      }
    }
  />
  <div
    className="makeStyles-root"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      hello world
    </h1>
    <div
      className="makeStyles-content"
    >
      <div
        className="makeStyles-item"
      >
        <p
          className="MuiTypography-root label MuiTypography-body1"
        >
          common:marketCap
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          N/A
        </p>
      </div>
      <div
        className="makeStyles-item"
      >
        <p
          className="MuiTypography-root label MuiTypography-body1"
        >
          common:inflation
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          0%
        </p>
      </div>
      <div
        className="makeStyles-item"
      >
        <p
          className="MuiTypography-root label MuiTypography-body1"
        >
          common:apr
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          0%
        </p>
      </div>
      <div
        className="makeStyles-item"
      >
        <p
          className="MuiTypography-root label MuiTypography-body1"
        >
          common:supply
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          0 
        </p>
      </div>
      <div
        className="makeStyles-item"
      >
        <p
          className="MuiTypography-root label MuiTypography-body1"
        >
          common:communityPool
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          0 
        </p>
      </div>
    </div>
  </div>
</div>
`;
